internal/controllers/mbsc_reconciler.go

```diff
@@ -0,0 +1,200 @@
+	}
+
+	err = r.reconHelperAPI.processImagesSpecs(ctx, mbscObj)
+	if err != nil {
+		return res, fmt.Errorf("failed to process images of MSBC %s: %v", mbscObj.Name, err)
+	}
+
+	logger.Info("run garbage collector for build/sign pods")
+	err = r.reconHelperAPI.garbageCollect(ctx, mbscObj)
+	if err != nil {
+		return res, fmt.Errorf("failed to run garbage collector for MBSC %s: %v", mbscObj.Name, err)
+	}
+
+	return res, nil
+}
+
+//go:generate mockgen -source=mbsc_reconciler.go -package=controllers -destination=mock_mbsc_reconciler.go mbscReconcilerHelperAPI
+
+type mbscReconcilerHelperAPI interface {
+	updateStatusByBuilds(ctx context.Context, mbscObj *kmmv1beta1.ModuleBuildSignConfig) error
+	processImagesSpecs(ctx context.Context, mbscObj *kmmv1beta1.ModuleBuildSignConfig) error
+	garbageCollect(ctx context.Context, mbscObj *kmmv1beta1.ModuleBuildSignConfig) error
+}
+
+type mbscReconcilerHelper struct {
+	client       client.Client
+	buildSignAPI buildsign.Manager
+	mbscAPI      mbsc.MBSC
+}
+
+func newMBSCReconcilerHelper(client client.Client, buildSignAPI buildsign.Manager, mbscAPI mbsc.MBSC) mbscReconcilerHelperAPI {
+	return &mbscReconcilerHelper{
+		client:       client,
+		buildSignAPI: buildSignAPI,
+		mbscAPI:      mbscAPI,
+	}
+}
+
+func (mrh *mbscReconcilerHelper) updateStatusByBuilds(ctx context.Context, mbscObj *kmmv1beta1.ModuleBuildSignConfig) error {
+	errs := make([]error, 0, len(mbscObj.Spec.Images))
+	patchFrom := client.MergeFrom(mbscObj.DeepCopy())
+	for _, imageSpec := range mbscObj.Spec.Images {
+		status, err := mrh.buildSignAPI.GetStatus(ctx, mbscObj.Name, mbscObj.Namespace, imageSpec.ModuleImageSpec.KernelVersion, imageSpec.Action, mbscObj)
+		errs = append(errs, err)
+		if err != nil || status == kmmv1beta1.BuildOrSignStatus("") {
+			// either we could not get the status or the status is empty
+			continue
+		}
+		mrh.mbscAPI.SetImageStatus(mbscObj, imageSpec.Image, imageSpec.Action, status)
+	}
+
+	err := mrh.client.Status().Patch(ctx, mbscObj, patchFrom)
+	errs = append(errs, err)
+	return errors.Join(errs...)
+}
+
+func (mrh *mbscReconcilerHelper) processImagesSpecs(ctx context.Context, mbscObj *kmmv1beta1.ModuleBuildSignConfig) error {
+	logger := log.FromContext(ctx)
+	errs := make([]error, 0, len(mbscObj.Spec.Images))
+	for _, imageSpec := range mbscObj.Spec.Images {
+		imageStatus := mrh.mbscAPI.GetImageStatus(mbscObj, imageSpec.Image, imageSpec.Action)
+		if imageStatus == kmmv1beta1.ActionSuccess {
+			// in case action succeeded - skip to the next image. Otherwise, action had not been handled yet, or is being handled, or already failed.
+			// in that case the Sync API will take care of what is needed to be done
+			continue
+		}
+		mld := createMLD(mbscObj, &imageSpec.ModuleImageSpec)
+		err := mrh.buildSignAPI.Sync(ctx, mld, true, mbscObj, imageSpec.Action)
+		errs = append(errs, err)
+		if err != nil {
```

Comment:
If we already have a `if err != nil` block then let's put the previous line in it.